### PR TITLE
[Improve][Connector] Improve write parquet

### LINKED ISSUE #2942: [Improve][Connector V2] Improve read & write parquet files
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

In MVP version writing data to parquet files not support complex types such as `list` `array` `map` `struct` etc...., in this pr it will support these complex data types when writing parquet files.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by TyrantLucifer**
## Purpose of this pull request

Improve write parquet  

close #2942

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by hailin0** [comment]
great pr

**Comment by TyrantLucifer** [comment]
> great pr

Pr is ready, pls review @hailin0 @Hisoka-X @EricJoy2048.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/ParquetWriteStrategy.java [hunk 1]
```diff
@@ -103,41 +152,142 @@ private ParquetWriter<GenericRecord> getOrCreateWriter(@NonNull String filePath)
...
     }
 
-    private Schema buildSchemaWithRowType() {
-        ArrayList<Schema.Field> fields = new ArrayList<>();
+    @SuppressWarnings("checkstyle:MagicNumber")
+    private Object resolveObject(Object data, SeaTunnelDataType<?> seaTunnelDataType) {
+        switch (seaTunnelDataType.getSqlType()) {
+            case ARRAY:
+                BasicType<?> elementType = ((ArrayType<?, ?>) seaTunnelDataType).getElementType();
+                ArrayList<Object> records = new ArrayList<>();
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/ParquetWriteStrategy.java]: Gave an init size is better.

Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sink/writer/ParquetWriteStrategy.java]: Resolved.

**Review by EricJoy2048 — APPROVED** [review]
+1

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**